plugins/tpm-eventlog/fu-tpm-eventlog.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN				"FuTpmEventlog"
+
+#include "config.h"
+
+#include <fwupd.h>
+#include <glib/gi18n.h>
+#include <locale.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "fu-tpm-eventlog-parser.h"
+
+static gboolean
+fu_tmp_eventlog_process (const gchar *fn, GError **error)
+{
+	gsize bufsz = 0;
+	g_autofree guint8 *buf = NULL;
+	g_autoptr(GPtrArray) items = NULL;
+	g_autoptr(GString) str = g_string_new (NULL);
+
+	/* parse this */
+	if (!g_file_get_contents (fn, (gchar **) &buf, &bufsz, error))
+		return FALSE;
+	items = fu_tpm_eventlog_parser_new (buf, bufsz,
+					    FU_TPM_EVENTLOG_PARSER_FLAG_ALL_ALGS |
+					    FU_TPM_EVENTLOG_PARSER_FLAG_ALL_PCRS,
+					    error);
+	if (items == NULL)
+		return FALSE;
+	for (guint i = 0; i < items->len; i++) {
+		FuTpmEventlogItem *item = g_ptr_array_index (items, i);
+		fu_tpm_eventlog_item_to_string (item, 0, str);
+		g_string_append (str, "\n");
+	}
+
+	/* success */
+	g_print ("%s", str->str);
+	return TRUE;
+}
+
+int
+main (int argc, char *argv[])
+{
+	gboolean verbose = FALSE;
+	gint rc = EXIT_SUCCESS;
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GOptionContext) context = g_option_context_new (NULL);
+	const GOptionEntry options[] = {
+		{ "verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose,
+			/* TRANSLATORS: command line option */
+			_("Show extra debugging information"), NULL },
```

Comment:
I think it's still useful to be able to only look at one PCR log at a time.  Showing the whole log gets confusing because depending on the order of extend events you might have 3 events for PCR 0 here and then some for PCR 7 and then another for PCR 0 etc.

Reply:
Could we just sort them by the PCR index?

Reply:
I added the sorting, it helps a lot.

Reply:
Yeah sorting certainly helps.  However an option to use is cheap, take a look at the fixup commit I just added.